src/herder/HerderImpl.cpp

```diff
@@ -2387,6 +2424,27 @@ HerderImpl::purgeOldPersistedTxSets()
     }
 }
 
+void
+HerderImpl::startCheckForDeadNodesInterval()
+{
+    LocalNode::forAllNodes(getSCP().getLocalNode()->getQuorumSet(),
+                           [this](const NodeID& nodeId) {
+                               mMissingNodes.insert(KeyUtils::toStrKey(nodeId));
```

Comment:
Do we need to do this? We could also use unmodified mMissingNodes.

Reply:
Is this asking why we populate the set or why we use `KeyUtils::toStrKey`?

We populate the set so that over the 15 minute interval, `mMissingNodes` contains only the intersection of the nodes that have been missing during the whole period.

We use `toStrKey` because the SCP subsystem gives us the missing information with the node ids as strings.

Reply:
(Updating to use NodeIDs)

Reply:
I see. I was wondering about the latter. I think this is related to my other comment about mapping public keys to node IDs. Yeah, storing NodeIDs in Herder is definitely cleaner. You could add a helper function to SCP to return a set of missing NodeIDs. Otherwise, you could convert strings to NodeIDs every time the timer expires, but that's less clean.